include/envoy/stream_info/stream_info.h

```diff
@@ -303,6 +307,11 @@ class StreamInfo {
    */
   virtual absl::optional<uint32_t> responseCode() const PURE;
 
+  /**
+   * @return the original port from the authority header iff it was removed by Envoy.
+   */
+  virtual absl::optional<uint32_t> removedHostPort() const PURE;
```

Comment:
meta question but does it make sense to put stuff like this into FilterState so that we can avoid bloating StreamInfo with more and more stuff?